mpf/devices/segment_display/segment_display.py

```diff
@@ -155,11 +165,56 @@ def add_text_entry(self, text, color, flashing, flash_mask, transition, transiti
             raise ValueError(f"Unknown update_method '{self.config['update_method']}' for segment display {self.name}")
 
         # For the replace-text update method, skip the stack and write straight to the display
-        new_text = TextTemplate(self.machine, text).evaluate({})
-        text = SegmentDisplayText.from_str(new_text, self.size, self.config['integrated_dots'],
-                                           self.config['integrated_commas'], self.config['use_dots_for_commas'],
-                                           color)
-        self._update_display(SegmentDisplayState(text, flashing, flash_mask))
+
+        ###############################
+        #Store current color and text as previous text/color of next run even if no transition in this step,
+        #the next step might have a transition, that the old text/color needs to be included into that transition
+        ###############################
+
+        # Handle new and previous text
+        if self._previous_text:
+            previous_text = self._previous_text
+        else:
+            previous_text = ""
+        self._previous_text = text  # Save the new text as the next previous text
+
+        # Handle new and previous color
+        if self._previous_color:
+            previous_color = self._previous_color
+        else:
+            previous_color = self._default_color
+        self._previous_color = color  # Save the new color as the next previous color
+
+        if transition or self._previous_transition_out:
+            if transition: #if transition exists, then ignore transition_out of previous text/color
```

Comment:
This second nested block could be removed by passing both possible values as arguments.
```
if transition or self._previous_transition_out:
    transition_conf = TransitionManager.get_transition(self.size,
                                                       self.config['integrated_dots'],
                                                       self.config['integrated_commas'],
                                                       self.config['use_dots_for_commas'],
                                                       transition or self._previous_transition_out)
    self._previous_transition_out = transition_out or None
```